geomesa-accumulo/geomesa-accumulo-datastore/src/test/scala/org/locationtech/geomesa/process/query/KNearestNeighborSearchProcessTest.scala

```diff
@@ -0,0 +1,209 @@
+ * Copyright (c) 2013-2019 Commonwealth Computer Research, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, Version 2.0
+ * which accompanies this distribution and is available at
+ * http://www.opensource.org/licenses/apache2.0.php.
+ ***********************************************************************/
+
+package org.locationtech.geomesa.process.query
+
+import org.geotools.data.Query
+import org.geotools.data.simple.SimpleFeatureCollection
+import org.geotools.feature.DefaultFeatureCollection
+import org.geotools.filter.text.ecql.ECQL
+import org.junit.runner.RunWith
+import org.locationtech.geomesa.accumulo.TestWithDataStore
+import org.locationtech.geomesa.features.ScalaSimpleFeature
+import org.locationtech.geomesa.utils.collection.SelfClosingIterator
+import org.locationtech.geomesa.utils.geohash.VincentyModel
+import org.locationtech.geomesa.utils.geotools.Conversions._
+import org.locationtech.geomesa.utils.geotools.SimpleFeatureTypes
+import org.opengis.feature.simple.SimpleFeature
+import org.specs2.runner.JUnitRunner
+
+import scala.util.Random
+
+@RunWith(classOf[JUnitRunner])
+class KNearestNeighborSearchProcessTest extends TestWithDataStore {
+
+  sequential
+
+  override val spec: String = "label:String,*geom:Point:srid=4326"
+  override val dtgField: Option[String] = None
+
+  val knn = new KNearestNeighborSearchProcess()
+
+  val rng = new Random(0)
+
+  val centerLat = 38.149894
+  val centerLon = -79.073639
+  val width = 0.30
+
+  val uvaLawn = Seq(
+    ScalaSimpleFeature.create(sft, "rotunda", "cville", "POINT( -78.503547 38.035475 )"),
+    ScalaSimpleFeature.create(sft, "pavilion I", "cville", "POINT( -78.503923 38.035536 )"),
+    ScalaSimpleFeature.create(sft, "pavilion II", "cville", "POINT( -78.503109 38.035278 )"),
+    ScalaSimpleFeature.create(sft, "pavilion III", "cville", "POINT( -78.504059 38.035308 )"),
+    ScalaSimpleFeature.create(sft, "pavilion IV", "cville", "POINT( -78.503180 38.035039 )"),
+    ScalaSimpleFeature.create(sft, "pavilion V", "cville", "POINT( -78.504276 38.034971 )"),
+    ScalaSimpleFeature.create(sft, "pavilion VI", "cville", "POINT( -78.503424 38.034721 )"),
+    ScalaSimpleFeature.create(sft, "pavilion VII", "cville", "POINT( -78.504424 38.034628 )"),
+    ScalaSimpleFeature.create(sft, "pavilion VIII", "cville", "POINT( -78.503601 38.034343 )"),
+    ScalaSimpleFeature.create(sft, "pavilion IX", "cville", "POINT( -78.504617 38.034208 )"),
+    ScalaSimpleFeature.create(sft, "pavilion X", "cville", "POINT( -78.503833 38.033938 )"),
+    ScalaSimpleFeature.create(sft, "cabell", "cville", "POINT( -78.505152 38.032704 )"),
+    ScalaSimpleFeature.create(sft, "beams", "cville", "POINT( -78.510295 38.034283 )"),
+    ScalaSimpleFeature.create(sft, "mccormick", "cville", "POINT( -78.522288 38.032844 )"),
+    ScalaSimpleFeature.create(sft, "hep", "cville", "POINT( -78.520019 38.034511 )")
```

Comment:
Hehehehe.... did you know someone who worked in High Energy Physics?

Reply:
Hahaha you can tell I was still very excited about being back in town :)

Reply:
@jahhulbert-ccri take a gander at this IT test :)